Scaffolding fixes

qbikez opened the pull request:
Fixes and tests for DbContext scaffolding.

I see most of the changes are already included in the source, but I also added some tests for verifying generated context.

qbikez committed:
fix FbDesignTimeServices for use with ef context scaffolding

* use same namespace in FbDesignTimeServices  as everywhere else
* add FbOptions to serviceCollection in FbDesignTimeServices

qbikez committed:
Add scaffolding test for testing EF DbContext scaffold

qbikez committed:
Fix type mapping for FB version < 3 with dialect version >= 3

qbikez committed:
scaffolding: better context generation

* handle indices without fields
* handle tables without PK
* fix nullable columns (NULL switched with NOT NULL)
* better foreign key detection

qbikez committed:
Merge remote-tracking branch 'upstream/dev' into scaffolding-fixes

qbikez committed:
scaffolding: add scaffold.ps1 for manual scaffolding testing

qbikez committed:
fix failing unit test

After changing Book<->Author relation to many-to-many, inserted items count changed.

ralmsdeveloper commented:
There is something breaking the test, the Author Column was not found!

I'll see what's going on.

qbikez committed:
use different db file in ScaffoldingTest to avoid file access errors

xunit allows different test classes to run in parallel, which caused
file access errors on TestContext

qbikez commented:
There were some file access issues with "EFCoreSample.fdb", I changed the new class to use a different file name.

ralmsdeveloper commented:
@qbikez Thanks!
